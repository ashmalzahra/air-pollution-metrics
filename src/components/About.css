.about {
  padding: 1rem;
  font-size: 1.2rem;
  line-height: 1.8rem;
  text-align: justify;
  font-weight: 400;
  background-color: rgb(39, 46, 74);
  color: #5a9dffc2;
  padding-top: 16%;
}
