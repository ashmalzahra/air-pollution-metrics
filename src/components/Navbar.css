nav {
  background-color: rgb(39, 46, 74);
  color: #5a9dffc2;
  background-image: linear-gradient(rgb(39, 46, 74) 0%, rgb(21, 27, 39) 100%);
  font-size: 1rem;
  position: fixed;
  width: 100%;
  top: 0;
  left: 0;
  box-sizing: border-box;
}

.nav-icon {
  font-size: 1.5rem;
}

.nav {
  list-style: none;
  padding: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 0;
}

a {
  color: #5a9dffc2;
}
