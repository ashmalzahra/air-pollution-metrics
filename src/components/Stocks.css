
.main-section {
  height: 12rem;
  font-weight: bold;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  background-color: rgb(39, 46, 74);
  color: #5a9dffc2;
}

.top-icon {
  height: 100%;
  font-size: 7rem;
  display: flex;
  justify-content: center;
  align-items: center;
}

.title {
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.text {
  font-size: 1.8rem;
}

.heading {
  font-size: 1rem;
  font-weight: bold;
  padding: 0.5rem 1rem;
  background-color: rgb(39, 46, 74);
  background-image: linear-gradient(rgb(39, 46, 74) 0%, rgb(21, 27, 39) 100%);
  color: #5a9dffc2;
}

.symbol {
  color: #5a9dffc2;
  text-decoration: none;
}

.icon {
  font-size: 3rem;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
}

.stocks {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
}

.stock {
  height: 10rem;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  line-height: 1.6rem;
  background-color: rgb(39, 46, 74);
  color: #5a9dffc2;
}

.stock:nth-child(4n + 2),
.stock:nth-child(4n + 3):not(:first-child) {
  background-image: linear-gradient(rgb(39, 46, 74) 0%, rgb(21, 27, 39) 100%);
}

.arrow {
  text-align: right;
  padding: 0.5rem;
}

.topHalf {
  box-sizing: border-box;
  height: 5rem;
  display: grid;
  grid-template-columns: 4fr 1fr;
}

.bottomHalf {
  box-sizing: border-box;
  height: 5rem;
  font-weight: bold;
  display: flex;
  flex-direction: column;
  align-items: end;
  justify-content: flex-end;
  text-align: right;
  padding: 0.5rem;
}

#searchFormContainer {
  padding: 0.5rem;
  box-sizing: border-box;
  width: 100vw;
  position: fixed;
  top: 0;
  left: 0;
  background-color: rgb(39, 46, 74);
}

#searchForm {
  display: grid;
  grid-template-columns: 4fr 1fr;
  height: 100%;
  align-items: center;
  justify-content: center;
}

.form-control {
  padding: 0.5rem;
  font-style: italic;
  border: none;
  background: linear-gradient(45deg, black, transparent);
  color: white;
}

#closeSearch {
  color: #5a9dffc2;
  display: flex;
  justify-content: center;
  align-items: center;
}
