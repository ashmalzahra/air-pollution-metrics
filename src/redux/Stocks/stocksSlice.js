import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const url = 'https://financialmodelingprep.com/api/v3/stock-screener?marketCapMoreThan=1000000000&betaMoreThan=1&volumeMoreThan=10000&sector=Technology&exchange=NASDAQ&dividendMoreThan=0&limit=100&apikey=0b2a0309c6e564c4b179292f7eda7c80';

const initialState = {
  stocks: [],
  ifSucceed: false,
  isLoading: false,
  error: null,
};

const fetchStocks = createAsyncThunk(
  'stocks/fetchStocks',
  async () => {
    try {
      const { data } = await axios.get(url);
      console.log(data);
      return data;
    } catch (error) {
      return error.message;
    }
  },
);

const stocksSlice = createSlice({
  name: 'stocks',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder
      .addCase(fetchStocks.pending, (state) => ({
        ...state,
        isLoading: true,
      }))
      .addCase(fetchStocks.fulfilled, (state, action) => ({
        ...state,
        isLoading: false,
        ifSucceed: true,
        stocks: action.payload,
      }))
      .addCase(fetchStocks.rejected, (state) => ({
        ...state,
        isLoading: false,
      }));
  },
});

export { fetchStocks }
export default stocksSlice.reducer;