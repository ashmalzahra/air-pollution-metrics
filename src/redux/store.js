import { configureStore } from '@reduxjs/toolkit';
import homeReducer from './Home/homeSlice'

store = configureStore({
  reducer: {
    home: homeReducer,
  },
});

export default store