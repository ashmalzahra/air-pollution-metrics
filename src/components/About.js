import './About.css'

const About = () => {
    return(
        <section className="about">
            <p>
            We use Stock Ticker search lookup API. Search stocks that are in our API by company name or ticker. You can also use the exchange filter to narrow down your results. Company name, currency, and exchange are some of the fields that are returned by API.
            </p>
            <p>
            Stock screener is a more advanced way to search for stocks as there are numerous parameters such as market cap, price, volume, beta, and so on. For example, you can use this endpoint to find NASDAQ-listed software companies that pay dividends and have good liquidity.
            </p>
        </section>
    )
}

export default About