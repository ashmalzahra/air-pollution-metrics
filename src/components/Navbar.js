import { NavLink, useParams } from 'react-router-dom';
import { AiOutlineHome, AiOutlineKey, AiOutlineLeft, AiOutlineQuestionCircle } from "react-icons/ai";
import './Navbar.css'

const Navbar = () => {
    const handleOpenSearch = () => {
      const FormContainer = document.getElementById('searchFormContainer');
      FormContainer.classList.replace('hide', 'show');
    };
  
    let links = [];
    if (Object.keys(useParams()).length === 0) {
      links = [
        { path: '/', text: <AiOutlineHome />, key: 1 },
        { path: '/', text: 'MARKET CAP', key: 2, className:"nav-text" },
        {
          path: '/',
          text: <AiOutlineKey onClick={handleOpenSearch} />,
          key: 3,
        },
      ];
    } else {
      links = [
        { path: '/', text: <AiOutlineLeft />, key: 1 },
        { path: '/', text: 'COMPANY', key: 2 },
        { path: 'about', text: <AiOutlineQuestionCircle />, key: 3 },
      ];
    }
  
    return (
      <nav>
        <ul className='nav'>
          {links.map((link) => (
            <li key={link.key}>
              <NavLink to={link.path} className="nav-icon">{link.text}</NavLink>
            </li>
          ))}
        </ul>
      </nav>
    );
  };
  
  export default Navbar;