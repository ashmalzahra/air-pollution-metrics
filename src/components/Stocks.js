import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchStocks } from '../redux/Stocks/stocksSlice';
import { NavLink} from 'react-router-dom';

const Stocks = () => {
    const dispatch = useDispatch()
    const stocksArray = useSelector((store) => store.stocks.stocks)
    const ifSucceed = useSelector((store) => store.stocks.ifSucceed)

    useEffect(() => {
        if(stocksArray.length === 0){
            dispatch(fetchStocks())
        }
    }, [dispatch, ifSucceed, stocksArray])

    return(
        <main>
            <section>
                <h1>
                    Market Cap
                </h1>
                <div>
                    {stocksArray.map((stock) => 
                    <NavLink to={`stock/${stock.symbol}`}>
                    <div>
                    {stock.companyName}
                    <br/>
                    Market Cap: {stock.marketCap}
                    <br/>
                    </div>
                    </NavLink>
                    )}       
                </div>
            </section>
        </main>
    )
}

export default Stocks