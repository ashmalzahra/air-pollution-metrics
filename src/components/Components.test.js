import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Stocks from './Stocks';
import Stock from './StockDetails';
import Navbar from './Navbar';
import About from './About';

describe('Components Test', () => {
  test('Homepage Test', () => {
    const configureStore = store;
    const component = renderer.create(
      <Provider store={configureStore}>
        <MemoryRouter>
          <Stocks />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Detail Page Test', () => {
    const configureStore = store;
    const component = renderer.create(
      <Provider store={configureStore}>
        <MemoryRouter>
          <Stock />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('About Page Test', () => {
    const configureStore = store;
    const component = renderer.create(
      <Provider store={configureStore}>
        <MemoryRouter>
          <About />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('NavBar Test', () => {
    const configureStore = store;
    const component = renderer.create(
      <Provider store={configureStore}>
        <MemoryRouter>
          <Navbar />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
