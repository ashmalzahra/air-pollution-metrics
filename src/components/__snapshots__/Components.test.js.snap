// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Test About Page Test 1`] = `
<section
  className="about"
>
  <p>
    We use Stock Ticker search lookup API. Search stocks that are in our API by company name or ticker. You can also use the exchange filter to narrow down your results. Company name, currency, and exchange are some of the fields that are returned by API.
  </p>
  <p>
    Stock screener is a more advanced way to search for stocks as there are numerous parameters such as market cap, price, volume, beta, and so on. For example, you can use this endpoint to find NASDAQ-listed software companies that pay dividends and have good liquidity.
  </p>
</section>
`;

exports[`Components Test Detail Page Test 1`] = `
<section>
  <header>
    <div
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 2a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v12h.5a.5.5 0 0 1 0 1H.5a.5.5 0 0 1 0-1H1v-3a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v3h1V7a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v7h1V2zm1 12h2V2h-2v12zm-3 0V7H7v7h2zm-5 0v-3H2v3h2z"
        />
      </svg>
    </div>
    <h1
      className="name"
    />
  </header>
  <div
    className="breakdown"
  >
    <span>
      DATA BREAKDOWN
    </span>
  </div>
  <ul
    className="list"
  >
    <li
      className="item even"
    >
      <span>
        SYMBOL
      </span>
      <span>
        <span
          className="arrow"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M666.7 505.5l-246-178A8 8 0 0 0 408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
            />
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
      </span>
    </li>
    <li
      className="item odd"
    >
      <span>
        MARKET CAP
      </span>
      <span>
        $
        <span
          className="arrow"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M666.7 505.5l-246-178A8 8 0 0 0 408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
            />
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
      </span>
    </li>
    <li
      className="item even"
    >
      <span>
        BETA
      </span>
      <span>
        <span
          className="arrow"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M666.7 505.5l-246-178A8 8 0 0 0 408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
            />
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
      </span>
    </li>
    <li
      className="item odd"
    >
      <span>
        PRICE
      </span>
      <span>
        $
        <span
          className="arrow"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M666.7 505.5l-246-178A8 8 0 0 0 408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
            />
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
      </span>
    </li>
    <li
      className="item even"
    >
      <span>
        VOLUME
      </span>
      <span>
        <span
          className="arrow"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M666.7 505.5l-246-178A8 8 0 0 0 408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
            />
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
      </span>
    </li>
  </ul>
</section>
`;

exports[`Components Test Homepage Test 1`] = `
Array [
  <main>
    <section
      className="main-section"
    >
      <div
        className="top-icon"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm-600-80h56c4.4 0 8-3.6 8-8V560c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v144c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V384c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v320c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V462c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v242c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V304c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v400c0 4.4 3.6 8 8 8z"
          />
        </svg>
      </div>
      <div
        className="title"
      >
        <div>
          <span
            className="text"
          >
            TOTAL
          </span>
          <br />
          <span
            className="value"
          >
            $
            0
          </span>
        </div>
      </div>
    </section>
    <section
      className="heading"
    >
      STATS BY COMPANY
    </section>
    <section
      className="stocks"
    >
      NO COMPANIES FOUND
    </section>
  </main>,
  <section
    className="hide"
    id="searchFormContainer"
  >
    <form
      className=""
      id="searchForm"
      onSubmit={[Function]}
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Search company ticker"
        type="text"
      />
      <div
        id="closeSearch"
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 0 0-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
          />
          <path
            d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
        </svg>
      </div>
    </form>
  </section>,
]
`;

exports[`Components Test NavBar Test 1`] = `
<nav>
  <ul
    className="nav"
  >
    <li>
      <a
        aria-current="page"
        className="nav-icon active"
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        className="nav-icon active"
        href="/"
        onClick={[Function]}
      >
        MARKET CAP
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        className="nav-icon active"
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M608 112c-167.9 0-304 136.1-304 304 0 70.3 23.9 135 63.9 186.5l-41.1 41.1-62.3-62.3a8.15 8.15 0 0 0-11.4 0l-39.8 39.8a8.15 8.15 0 0 0 0 11.4l62.3 62.3-44.9 44.9-62.3-62.3a8.15 8.15 0 0 0-11.4 0l-39.8 39.8a8.15 8.15 0 0 0 0 11.4l62.3 62.3-65.3 65.3a8.03 8.03 0 0 0 0 11.3l42.3 42.3c3.1 3.1 8.2 3.1 11.3 0l253.6-253.6A304.06 304.06 0 0 0 608 720c167.9 0 304-136.1 304-304S775.9 112 608 112zm161.2 465.2C726.2 620.3 668.9 644 608 644c-60.9 0-118.2-23.7-161.2-66.8-43.1-43-66.8-100.3-66.8-161.2 0-60.9 23.7-118.2 66.8-161.2 43-43.1 100.3-66.8 161.2-66.8 60.9 0 118.2 23.7 161.2 66.8 43.1 43 66.8 100.3 66.8 161.2 0 60.9-23.7 118.2-66.8 161.2z"
          />
        </svg>
      </a>
    </li>
  </ul>
</nav>
`;
